runtime/src/test/java/com/flipkart/flux/dao/StateMachinesDAOTest.java

```diff
@@ -13,56 +13,87 @@
-import com.flipkart.flux.guice.module.ConfigModule;
-import com.flipkart.flux.guice.module.HibernateModule;
-import com.google.inject.Guice;
-import com.google.inject.Injector;
-
+import com.flipkart.flux.representation.DomainTypeCreator;
+import com.flipkart.flux.rules.DbClearRule;
+import com.flipkart.flux.runner.GuiceJunit4Runner;
 import junit.framework.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
 
+import javax.inject.Inject;
 import javax.transaction.Transactional;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * <code>StateMachinesDAOTest</code> class tests the functionality of {@link StateMachinesDAO} using JUnit tests.
  * @author shyam.akirala
  * @author kartik.bommepally
  */
+@RunWith(GuiceJunit4Runner.class)
 public class StateMachinesDAOTest {
 
-    private Injector injector;
+    @Inject
+    @Rule
+    public DbClearRule dbClearRule;
+
+    @Inject
+    StateMachinesDAO stateMachinesDAO;
+
+    @Inject
+    StatesDAO statesDAO;
+
+    @Inject
+    DomainTypeCreator domainTypeCreator;
 
     @Before
-    public void setup() {
-        injector = Guice.createInjector(new ConfigModule(), new HibernateModule());
-    }
+    public void setup() {}
 
-    @Test
-    @Transactional
+    @Test @Transactional
     public void createSMTest() {
-        StateMachinesDAO stateMachinesDAO = injector.getInstance(StateMachinesDAO.class);
-        StatesDAO statesDAO = injector.getInstance(StatesDAO.class);
         DummyTask task = new DummyTask();
         DummyOnEntryHook onEntryHook = new DummyOnEntryHook();
         DummyOnExitHook onExitHook = new DummyOnExitHook();
-        State<DummyEventData> state1 = statesDAO.create(new State<DummyEventData>(2L, "state1", "desc1", onEntryHook, task, onExitHook, 3L, 60L));
-        State<DummyEventData> state2 = statesDAO.create(new State<DummyEventData>(2L, "state2", "desc2", null, null, null, 2L, 50L));
-        Set<State<DummyEventData>> states = new HashSet<State<DummyEventData>>();
+        State<DummyEventData> state1 = statesDAO.create(new State<>(2L, "state1__", "desc1", onEntryHook, task, onExitHook, 3L, 60L));
+        State<DummyEventData> state2 = statesDAO.create(new State<>(2L, "state2__", "desc2", null, null, null, 2L, 50L));
+        Set<State<DummyEventData>> states = new HashSet<>();
         states.add(state1);
         states.add(state2);
-        StateMachine<DummyEventData> stateMachine = new StateMachine<DummyEventData>(2L, "test_name", "test_desc", states);
+        StateMachine<DummyEventData> stateMachine = new StateMachine<>(2L, "SM_name_", "SM_desc_", states);
         String savedSMId = stateMachinesDAO.create(stateMachine).getId();
 
         StateMachine stateMachine1 = stateMachinesDAO.findById(savedSMId);
         Assert.assertNotNull(stateMachine1);
     }
+
+    /** Test creation of State machine from state machine definition */
+    @Test @SuppressWarnings("unchecked")
```

Comment:
Wanted to avoid the warnings which are getting raised due to missing type parameters. Added type parameters in all places and removed suppress warnings. @yogeshnachnani